### ISSUE TITLE
Multipl MySensors Gateway with MQTT interface - same client ID - disconnection

### ISSUE THREAD

**Issue by darkManPi** [domoticz/domoticz#3793]
There is still problem with MQTT Mysensors connection.
With more than one  "MySensors Gateway with MQTT interface" hardwares connecting to same MQTT server, MQTT will keep on disconnecting/reconnecting. 

There was similar case #3750 
In fact commit b89ba87 help for case with single MQTT connection per Domoticz instance having  few independent domoticz instances, but not for above case.

**How to simulate problem:**

Create 2  "MySensors Gateway with MQTT interface" hardware connected to the same MQTT server. "Topic Prefix Mode" have not  influence.
After enabling them watch domoticz log.
Both hardware will keep on "fighting" for MQTT connection and MQTT server will be simply disconnecting one by one:

```1576152152: Client Domoticz-MySensorsb9bc2348-483f-4e3a-9b0f-f80129080239 already connected, closing old connection.
1576152152: Client Domoticz-MySensorsb9bc2348-483f-4e3a-9b0f-f80129080239 disconnected.
1576152152: New client connected from 192.168.221.5 as Domoticz-MySensorsb9bc2348-483f-4e3a-9b0f-f80129080239 (c1, k40).
1576152178: New connection from ....
```

**Solution:**

Generate unique MQTT client ID for each single Mys MQTT connection. At current moment MysMQTT client id is global for whole Domoticz instance as  introduced in commit b89ba87 / c55e29c.
"szRandomUUID = GenerateUUID();" could be set for example inside hardware/MySensorsMQTT.cpp. 
Same issue is exist for "hardware/TTNMQTT.cpp".

Greetings

**Comment by gizmocuz** [comment]
@darkManPi , please do not create new issue topics... and make a PR with your change please ;)

**Comment by rwaaren** [comment]
@darkManPi  Should be solved in Beta >= V4.11569 . Can you please check and report back

**Comment by darkManPi** [comment]
> .. please do not create new issue topics... and make a PR with your change please ;)
I had the mod ready at time of writing the comment.. But the environment for domoticz compilation was not ready yet. I prefer not to push something not tested.

> ..Should be solved in Beta >= V4.11569 . Can you please check and report back
It is working well now. 
For me we can close the point.

**Comment by gizmocuz** [comment]
Thanks all !!
